### ISSUE TITLE
Schema fails to correctly list custom settings fields

### ISSUE THREAD

**Issue by OWMC** [wp-graphql/wp-graphql#1925]
**Expected Behaviour**
When I have a custom setting in WP,
the field names should be listed under the group names in the WPGraphQL IDE,
so that I can query those fields.

**Current Behaviour**
If the custom setting group option name contains an underscore,
the fields belonging to that group do not show up under the group name in the WPGraphQL IDE,
and I cannot query those fields (unless I use the "allSettings" group),
and i cannot build a Gatsby (v2.25.3) site using gatsby-source-graphql (v2.7.7) because `Error: {TYPENAME}_{customGroupNameSetting} must define one or more fields`.

**Possible Solution**
A workaround is to simply not use underscores in the names of custom settings group names (I'm using camel case).

**Steps to Reproduce**

- Install a fresh instance of WP (v5.7.2),
- Add a custom setting to the theme's functions.php (https://www.wpgraphql.com/docs/settings/#registering-custom-settings)

```
/**
 * Registers a text field setting for WordPress 4.7 and higher.
 **/
function register_my_setting() {
    $args = [
      'type' => 'string',
      'sanitize_callback' => 'sanitize_text_field',
      'default' => NULL,
      'show_in_graphql' => true, // This tells WPGraphQL to show this setting in the Schema
    ];
    register_setting( 'my_options_group', 'my_option_name', $args );
}

add_action( 'init', 'register_my_setting' );
```

- Add WPGraphQL v1.3.8 and activate
- View WPGraphQL IDE to try and query the custom setting group and fields

**Context (Environment)**
Locally (and anywhere)

**Comment by jasonbahl** [comment]
@OWMC thanks for opening this issue and providing details to reproduce!

**Comment by jasonbahl** [comment]
@OWMC I've reproduced this behavior with the code you shared. 

I see the Schema has a Type: `MyOptionsGroupSettings` but the type has no fields: 

```graphql
query GetType {
  __type(name: "MyOptionsGroupSettings") {
    name
    kind
    fields {
      name
      type {
        kind
      }
    }
  }
}
```

![Screen Shot 2021-05-27 at 9 39 57 AM](https://user-images.githubusercontent.com/1260765/119855973-99a27280-becf-11eb-9905-3d0305cef7a7.png)

The expectation would be for the `MyOptionsGroupSettings` to have 1 field named `myOptionName` of the `String` type.

**Comment by jasonbahl** [comment]
I've confirmed that PR #1953 from @markkelnar fixes this issue. 

Using the code shared above: 

```php
function register_my_setting() {
	$args = [
		'type' => 'string',
		'sanitize_callback' => 'sanitize_text_field',
		'default' => NULL,
		'show_in_graphql' => true, // This tells WPGraphQL to show this setting in the Schema
	];
	register_setting( 'my_options_group', 'my_option_name', $args );
}

add_action( 'init', 'register_my_setting' );
```

I can then see the Option Group and the individual field in the Schema: 

![Screen Shot 2021-06-15 at 9 35 49 AM](https://user-images.githubusercontent.com/1260765/122082246-1339cc00-cdbd-11eb-8f64-0e5c772f7516.png)

And I can use GraphQL Introspection to see the Group and the Field as expected: 

![Screen Shot 2021-06-15 at 9 36 20 AM](https://user-images.githubusercontent.com/1260765/122082319-25b40580-cdbd-11eb-9ce2-da8f6e55a71c.png)

**Comment by jasonbahl** [comment]
Closed by #1953
